packages/slate-react/src/components/editable.tsx

```diff
@@ -392,9 +376,25 @@ export const Editable = (props: EditableProps) => {
         }
       }
     }, 100),
-    []
+    [readOnly]
```

Comment:
The debounce for this function will somehow need to be cleared too, otherwise it's possible it memoizes and uses the stale value while it's waiting during debouncing.

Reply:
[This PR](https://github.com/ianstormtaylor/slate/issues/3338) will fix that it looks like, so all good if both are merged.